surveys/views.py

```diff
@@ -43,12 +44,31 @@ class SurveySubmittedList(TemplateView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        surveys_submitted = Survey.objects.all().order_by('form_id', '-time_stop').distinct('form_id')
+        surveys = Survey.objects.all()
+        context['surveys_submitted'] = surveys.order_by('form_id', '-time_stop').distinct('form_id')
 
-        for survey in surveys_submitted:
+        for survey in context['surveys_submitted']:
             survey.form_title = FormEntry.objects.get(id=survey.form_id)
 
-        context['surveys_submitted'] = surveys_submitted
+        return context
+
+
+class SurveySubmittedDetail(TemplateView):
+    template_name = "survey_submitted_detail.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['form_entry'] = FormEntry.objects.get(id=context['form_entry_id'])
+        context['surveys_submitted'] = Survey.objects.filter(form_id=context['form_entry_id'])
+
+        for survey in context['surveys_submitted']:
+            row = SurveyRow.objects.get(survey=survey)
+            survey.row_total = row.total
+            survey.components = SurveyComponent.objects.filter(row=row).order_by('position')
```

Comment:
One alternative pattern that could be useful here would be to move these queries to property methods on the `Survey` model. That way if you wind up wanting to get the row or components for a Survey often, you can treat them like properties on the model instead of having to remember the queries for them each time. Something like:

```python
class Survey(db.model):
...
    @property
    def row(self):
        ...

    @property
    def components(self):
    ...
```

I think this is an optional step at this point, but it's something to keep in mind as you continue to flesh out the relationship between `Survey`s, `SurveyRow`s, and `SurveyComponents`.

Reply:
Ah nice I'd like to get more comfortable with this pattern! Here's a PR in the `django-pldp` repo to get it working: https://github.com/datamade/django-pldp/pull/10